### DISCUSSION TITLE
Adding JWT expiration to login route

### DISCUSSION THREAD

**Discussion by mlisthenewcool** [fastapi-users/fastapi-users#721] [Q&A] [answered]
Is there a mechanism to add token expiration time to HTTP response using the `[...]/login` route ?
I can see that `get_login_response` in https://github.com/fastapi-users/fastapi-users/blob/master/fastapi_users/authentication/jwt.py doesn't.

**Comment by frankie567** [comment, marked as answer]
The token lifetime is set when initializing `JWTAuthentication` class: https://fastapi-users.github.io/fastapi-users/configuration/authentication/jwt/#jwt

**Reply by mlisthenewcool** [reply]
I meant in HTTP response, in addition to `access_token` & `token_type`.

**Reply by frankie567** [reply]
You can access it by decoding the JWT. If I consider this JWT:

```
eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjoiMTIzNCIsImF1ZCI6WyJmYXN0YXBpLXVzZXJzOmF1dGgiXSwiZXhwIjoxNjMxMjAzMTcxfQ.t_oP0P2ssldYfeQs4QRTtsVk1iMfRG7Q8JBGa5VJcOk
```

It'll be decoded as this:

```json
{
  "user_id": "1234",
  "aud": [
    "fastapi-users:auth"
  ],
  "exp": 1631203171
}
```

where `exp` is the timestamp expiration time.

**Reply by mlisthenewcool** [reply]
Oh, I didn't read enough doc about JWT itself, thanks ! Solved my problem.
